homeassistant/components/fritz/common.py

```diff
@@ -175,15 +176,16 @@ def setup(self) -> None:
 
         self._update_available, self._latest_firmware = self._update_device_info()
 
-    async def async_start(self, options: MappingProxyType[str, Any]) -> None:
-        """Start FritzHosts connection."""
-        self.fritz_hosts = FritzHosts(fc=self.connection)
-        self._options = options
-        await self.hass.async_add_executor_job(self.scan_devices)
-
-        self._cancel_scan = async_track_time_interval(
-            self.hass, self.scan_devices, timedelta(seconds=TRACKER_SCAN_INTERVAL)
-        )
+    @callback
+    async def _async_update_data(self) -> None:
+        """Update FritzboxTools data."""
+        try:
+            self.fritz_hosts = FritzHosts(fc=self.connection)
+            self.scan_devices()
```

Comment:
This is sync methods while async update data is marked as a callback. You need to write a function to offload this to the executor.

Reply:
Indeed, thx for spotting.

Simone